### ISSUE TITLE
Restore custom notification sounds functionality ?

### ISSUE THREAD

**Issue by thestupidbox** [Jonius7/SteamUI-OldGlory#83]
Hello,

It used to be possible on Steam to change all of the notification sounds (Friends, Message etc...) by changing the sound files inside the "friends" and "resource" folder. Around 2018ish it seems that Valve made a launcher update where they not only changed the sounds, but they are also now loaded elsewhere _(probably serverside as I cannot locate the new sounds anywhere on the steam directory)_ even though the old sound files are still present. As of today, I think it's **impossible** to have custom notification sounds unless you downgrade Steam to an earlier version.

So, would it be possible to find a way to replace and load custom notification sounds or restore "the old way" of loading notification sounds ?

Thanks !

**Comment by Jonius7** [comment]
I had a look around and seems it isn't possible according to this: https://steamcommunity.com/discussions/forum/1/2789318172111960180/

Have you tried changing these files in `friends` or they don't change anything?
![2023-03-20 14_19_21-friends](https://user-images.githubusercontent.com/1049415/226238877-767f2e59-7af8-4bcb-b70a-e157c5cac896.png)

**Comment by thestupidbox** [comment]
I just tried it again on the latest Steam version with all of the sounds changed and set to read only and I found out that : 

- Steam doesn't care about the replaced sounds in the `friends` folder, it's still the new ones _(which are nowhere to be found on the steam folder)_ even after i carefully changed those. 

- The ones in the `resource` however folder still work as intended. Changing those sounds actually makes Steam use those sounds I modified. Which are the screenshot and the call sounds.

It seems like this has all to do with the chat update that **only** changed how the sounds in the friends menu are loaded so it's currently impossible to change those sounds unless someone can figure it out. Since there's a patch to modify the friends list css, maybe it's possible to make one that could bring back the old method of loading sounds in the friends list too.

**Comment by PhantomGamers** [comment]
You are correct that these are loaded server side now. They are loaded in the `friends.js` from https://community.akamai.steamstatic.com/public/javascript/webui/friends.js

With the sound files also being pulled from the web (e.g. `ui_steam_message_old_smooth.m4a` from https://community.akamai.steamstatic.com/public/sounds/webui/ui_steam_message_old_smooth.m4a )

These are then stored in Steam's chromium cache folder in `%localappdata%/Steam/htmlcache/Cache`

Sadly these are bundled into the `data_#` files which are collections that a bunch of smaller files get thrown into, and I've yet to figure out how to modify a specific file within them.

**Comment by thestupidbox** [comment]
Oh that's interesting info... I guess a workaround could be possible.

Since this is serverside, I personally first tried myself to have a janky solution using [Fiddler (Classic)](https://www.telerik.com/fiddler/fiddler-classic) intercept those URLs with my custom ones but no luck. After adding autorequests and deleting the cache Steam either plays no sounds or still insists to use the newer ones because I guess Fiddler can't intercept Steam in time :/

I browsed the cache on the `data_#` folder using [ChromeCacheView](https://www.nirsoft.net/utils/chrome_cache_view.html) and it did work, I could browse the cache files but what I also found interesting is the URLs that are linked to it. Which are different and could be the reason why my Fiddler solution didn't work. Just because the URL redirects were loaded from a different server.

![image](https://user-images.githubusercontent.com/128084237/226682217-219c26f5-9a27-4ea2-9622-79db6b7cfe7b.png)

So then I deleted the cache again, added new autorequests with this new URL and tried it again. This time Fiddler intercepted the URL and **the custom sounds worked !** It's by far not the ideal solution but it sure is a way to get the custom sounds manually saved and get used on Steam's Chromium cache.

**Comment by Jonius7** [comment]
So the question is, what can OldGlory do for you to help this process? AFAIK this is out of scope for something like OldGlory

**Comment by thestupidbox** [comment]
Well at first I thought that the way that determined how Steam loaded these sounds was locally stored and maybe something could've been done to patch and restore the way these sounds were loaded but it seems that now since the other `friends.js` _(not the local one in the `clientui` folder)_ is loaded seperately, this would be trickier to find a solution and then implement it in OldGlory. Maybe this is something that more likely could be implemented in [SFP](https://github.com/PhantomGamers/SFP) and make it possible to use custom sounds. I use both of these tools anyway.
